### PR TITLE
feat(project): add prettier-config-carbon package

### PR DESCRIPTION
**Pull request by joshblack**
Adds the `prettier-config-carbon` package to help have consistent prettier configuration across projects, both for Carbon and for teams building with Carbon.

#### Changelog

**New**

- Add `prettier-config-carbon`

**Changed**

- Update `@carbon/cli` to use new prettier configuration

**Removed**

### CODE REVIEW

**Review by emyarod — APPROVED** [review]
looks good to me

**Review by tay1orjones — COMMENTED** [review]
Should Prettier be specified as a peer dep? Is the intent here for this to be compatible with all versions of prettier, <=1, or >=2?

We're using Prettier v2, which has some new defaults. I don't think there's any changes to the way configs are set/used, just the config properties.

**Comment by joshblack** [comment]
Good call @tay1orjones! And I'd say for v2 but could easily have it apply to both 👀

**Review by aledavila — APPROVED** [review]

File: packages/prettier-config-carbon/package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "carbon",
+    "carbon-design-system",
+    "components",
+    "react"
+  ],
+  "publishConfig": {
+    "access": "public"
+  },
+  "peerDependencies": {
+    "prettier": "1x || 2.x"
```
Comment (joshblack) [inline packages/prettier-config-carbon/package.json]:
```suggestion
    "prettier": "1.x || 2.x"
```

File: packages/prettier-config-carbon/package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "prettier-config-carbon",
+  "description": "Prettier config for the Carbon Design System",
+  "version": "0.1.0",
```
Comment (joshblack) [inline packages/prettier-config-carbon/package.json]:
```suggestion
  "version": "0.0.0",
```

File: packages/cli/package.json [hunk 3]
```diff
@@ -30,6 +30,7 @@
     "fs-extra": "^8.0.1",
     "inquirer": "^6.4.1",
     "prettier": "^1.19.1",
+    "prettier-config-carbon": "^0.1.0",
```
Comment (joshblack) [inline packages/cli/package.json]:
```suggestion
    "prettier-config-carbon": "^0.0.0",
```

File: package.json [hunk 4]
```diff
@@ -67,6 +67,7 @@
     "lerna": "^3.20.2",
     "lint-staged": "^9.5.0",
     "prettier": "^1.19.1",
+    "prettier-config-carbon": "^0.1.0",
```
Comment (joshblack) [inline package.json]:
```suggestion
    "prettier-config-carbon": "^0.0.0",
```

**Pull request merged**